sycl/source/detail/device_image_impl.hpp

```diff
@@ -161,12 +161,29 @@ class device_image_impl {
 
   const context &get_context() const noexcept { return MContext; }
 
+  void set_kernel_ids(std::vector<kernel_id> KernelIDs) noexcept {
+    MKernelIDs = std::move(KernelIDs);
+  }
+
   std::vector<kernel_id> &get_kernel_ids_ref() noexcept { return MKernelIDs; }
 
   std::vector<unsigned char> &get_spec_const_blob_ref() noexcept {
     return MSpecConstsBlob;
   }
 
+  RT::PiMem &get_spec_const_buffer_ref() noexcept {
+    std::lock_guard<std::mutex> Lock{MSpecConstAccessMtx};
+    if (nullptr == MSpecConstsBuffer) {
+      const detail::plugin &Plugin = getSyclObjImpl(MContext)->getPlugin();
+      Plugin.call<PiApiKind::piMemBufferCreate>(
+          detail::getSyclObjImpl(MContext)->getHandleRef(),
+          PI_MEM_FLAGS_ACCESS_RW | PI_MEM_FLAGS_HOST_PTR_USE,
```

Comment:
Why do we use RW access here? I think, after the device image is built, we can't change spec const values. Can you at least put todo here, please?

Reply:
Changed to `PI_MEM_FLAGS_ACCESS_RO`